src/bundle/wysiwyg-preset.ts

```diff
@@ -63,9 +64,26 @@ export const BundlePreset: ExtensionAuto<BundlePresetOptions> = (builder, opts)
         baseSchema: {
             paragraphKey: f.toPM(A.Text),
             paragraphPlaceholder: (node: Node, parent?: Node | null) => {
-                const isDocEmpty =
-                    !node.text && parent?.type.name === BaseNode.Doc && parent.childCount === 1;
-                return isDocEmpty ? i18nPlaceholder('doc_empty') : null;
+                let isPlaceholderShown = false;
+
+                if (opts.placeholderOptions?.behavior === 'empty-row') {
+                    isPlaceholderShown = !node.text && parent?.type.name === 'doc';
+                } else {
+                    isPlaceholderShown =
+                        !node.text && parent?.type.name === BaseNode.Doc && parent.childCount === 1;
+                }
+
+                let text = i18nPlaceholder('doc_empty');
+
+                if (opts.placeholderOptions?.value) {
+                    if (typeof opts.placeholderOptions?.value === 'string') {
+                        text = opts.placeholderOptions?.value;
+                    } else {
+                        text = opts.placeholderOptions?.value();
+                    }
+                }
+
+                return isPlaceholderShown ? text : null;
```

Comment:
Let's rewrite this piece of code – make it more readable and a little more optimal

```js
const {behavior, value} = opts.placeholderOptions || {};

const isRowEmpty = !node.text && parent?.type.name === BaseNode.Doc;
const isDocEmpty = isRowEmpty && parent.childCount === 1;

const showPlaceholder = behavior === 'empty-row' ? isRowEmpty : isDocEmpty;

if (!showPlaceholder) return null;

return typeof value === 'function' ? value() : value ?? i18nPlaceholder('doc_empty');
```